### PR TITLE
feature: create update user profile form

### PR DESCRIPTION
**Pull request by halfangel37**
I created update user profile form , it have 2 fields to update is first name and last name (task 035) but I also do the image upload ( task 046 ) in it , can I do that to prepare for task 046 or I need to delete that part of code, and one more thing the vuetify grid system do not include the ipad pro so it seem pretty ugly , do I need to fix it.
I have finish the upload avatar feature and also add toast to the form.
Sorry but I forgot to push the code in the morning, please check again.

Updated Demo:

https://www.loom.com/share/52db00f5c0c04d5eaeae69a3e79e4073

upload image after choose file:
https://www.loom.com/share/cdbf2b9aa96149d8accb273546cc0132
Move toast to top-right:
image

### CODE REVIEW

File: src/profile/store/index.js [hunk 1]
```diff
@@ -1,14 +1,24 @@
 import { changeUserPassword } from "@/http/auth";
+import { httpClient } from "../../http/httpClient";
 import router from "../../router";
+import NProgress from "nprogress";
 export default {
   namespaced: true,
   state: {
     changePasswordMessage: "",
+    userName: { firstName: "", lastName: "" },
```
Comment (truonghungit) [inline src/profile/store/index.js:9]: ?? 
You should save all user properties to `user`. I don't know why the username content the first name & last name?

image

File: src/profile/store/index.js [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
   mutations: {
     SET_PASSWORD_MESSAGE(state, errorMessage) {
       state.changePasswordMessage = errorMessage;
     },
+    GET_USER_PROFILE(state, data) {
+      state.userName.firstName = data.firstName;
+      state.userName.lastName = data.lastName;
+    },
+    UPDATE_USER_AVATAR(data) {
+      console.log("image", data);
```
Comment (truonghungit) [inline src/profile/store/index.js:20]: remove console log,

File: src/profile/store/index.js [hunk 3]
```diff
@@ -26,8 +36,46 @@ export default {
           commit("SET_PASSWORD_MESSAGE", message);
         });
     },
+    getUserProfile({ commit }) {
+      return httpClient
+        .get("/accounts")
+        .then((response) => {
+          commit("GET_USER_PROFILE", response.data);
```
Comment (truonghungit) [inline src/profile/store/index.js:43]: I think `SET_USER_PROFILE` is better for this case

File: src/profile/store/index.js [hunk 4]
```diff
@@ -26,8 +36,46 @@ export default {
...
+        .catch((error) => {
+          console.log(error.response.data.errors);
+        });
+    },
+    updateUserProfile({ commit }, credentials) {
+      NProgress.start();
+      return httpClient
+        .put("/accounts", credentials)
+        .then((response) => {
+          commit("GET_USER_PROFILE", response.data);
```
Comment (truonghungit) [inline src/profile/store/index.js:54]: use `SET_USER_PROFILE`

File: src/profile/store/index.js [hunk 5]
```diff
@@ -26,8 +36,46 @@ export default {
...
+      console.log("before-commit:", formData);
+      NProgress.start();
+      return httpClient
+        .post("/accounts/avatar", formData)
+        .then((response) => {
+          commit("UPDATE_USER_AVATAR", response.data);
+          NProgress.done();
+        })
+        .catch((error) => {
+          console.log("error", error.response.data);
```
Comment (truonghungit) [inline src/profile/store/index.js:72]: remove console log

**Review by truonghungit — CHANGES_REQUESTED** [review]
image

Please check again

File: src/profile/store/index.js (hunk 1 above)
Comment (halfangel37) [reply src/profile/store/index.js:9]: image
in the put API link , we can only change 2 fields first name and last name , I could get username/email but cannot change it, so i think we just need 2 fields to update

**Review by luantc13 — CHANGES_REQUESTED** [review]
Nice work!

**Review by truonghungit — CHANGES_REQUESTED** [review]

**Review by luantc13 — APPROVED** [review]
Nice work!

**Review by truonghungit — APPROVED** [review]

**Review by dtngoc99 — APPROVED** [review]

**Pull request merged**